### PR TITLE
Add GROUP to functions 

### PR DESCRIPTION
**Pull request by jdunkerley**
### Pull Request Description

- Update list of groups to agreed list.
- Lower case `ALIAS` names to be consistent with function names.
- Add `GROUP` to methods.
- All constructors and functions have doc comments.
- Correct a few typos (e.g. `PRVIATE`).
- Mark some more things as `PRIVATE`.
- Use `ToDo:` and `Note:` consistently.
- Order tags in doc comment.

### Important Notes

We don't have all the doc comments on types and will want to add them in future,

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [ ] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

**Review by hubertp — APPROVED** [review]
:)

**Review by GregoryTravis — APPROVED** [review]

**Review by Akirathan — APPROVED** [review]

**Review by JaroslavTulach — APPROVED** [review]

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Interval.enso [hunk 1]
```diff
@@ -49,7 +48,8 @@ type Interval_Type
 
 ## A type representing an interval over real numbers.
 type Interval
-    ## Creates an interval.
+    ## GROUP DateTime
```
Comment (radeusgd) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Data/Interval.enso:51]: I assume spaces are not allowed in group names?

I find it slightly off to see `DateTime` where in all other places in Enso it would be `Date_Time`? - we always use Upper_Snake_Case and never CamelCase in Enso (apart from internal Java code).

Do you think we could have a space in the name?

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Numbers.enso [hunk 2]
```diff
@@ -348,7 +368,8 @@ type Number
     nan : Decimal
     nan = Double.NaN
 
-    ## Checks if the given number is the floating-point Not-a-Number value.
+    ## GROUP Logical
```
Comment (radeusgd) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Data/Numbers.enso]: Hmmm, wouldn't `is_nan` fit better into `Math` group?

It feels very similar to e.g. `is_infinite` that is there.

I assume a parallelism to `is_nothing`, but IMO `is_nothing` is a bit more special (and does not have a much better category anyway), whereas `is_nan` is rather closely related to the Math operations.

But that is still just my subjective suggestion, I think `Logical` seems OK too.

File: distribution/lib/Standard/AWS/0.0.0-dev/src/S3/S3.enso [hunk 3]
```diff
@@ -7,7 +7,8 @@ import project.Internal.Auth
 polyglot java import software.amazon.awssdk.services.s3.model.S3Exception
 polyglot java import software.amazon.awssdk.services.s3.S3Client
 
-## Gets the list of the S3 bucket names.
+## GROUP Standard.Base.Input
```
Comment (4e6) [inline distribution/lib/Standard/AWS/0.0.0-dev/src/S3/S3.enso:10]: Please update the _Tags_ section of the doc comments https://github.com/enso-org/enso/blob/d7e5c3f36274ab57edec649cec9578108ccaa31d/docs/syntax/comments.md#tags

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Pair.enso [hunk 4]
```diff
@@ -82,7 +84,8 @@ type Pair
             -2 -> self.first
             _ -> if_missing
 
-    ## Checks whether a predicate holds for at least one element of self pair.
+    ## GROUP Logical
+       Checks whether a predicate holds for at least one element of self pair.
```
Comment (radeusgd) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Data/Pair.enso]: Unrelated to this PR, but going thru the docs I see this `self` is an accidental artifact of the `this/self` keyword rename.

Let's fix it :)

```suggestion
       Checks whether a predicate holds for at least one element of this pair.
```

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Pair.enso [hunk 5]
```diff
@@ -47,15 +47,17 @@ type Pair
     map self fun =
         Pair.new (fun self.first) (fun self.second)
 
-    ## Returns the length (2) of the Pair.
+    ## GROUP Metadata
+       Returns the length (2) of the Pair.
```
Comment (radeusgd) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Data/Pair.enso]: unrelated, but shall we do?
```suggestion
       Returns the length of the Pair (which is always equal to 2).
```

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Range.enso [hunk 6]
```diff
@@ -37,7 +37,8 @@ type Range
          non-zero. Defaults to 1.
     Between (start : Integer) (end : Integer) (step : Integer = 1)
 
-    ## Creates a copy of this range with a changed step.
+    ## GROUP Metadata
```
Comment (radeusgd) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Data/Range.enso:40]: subjective, but I'm not sure `Metadata` fits here. Maybe it should be `Selections`?

If I have range `r1 = 0.up_to 100` and do `r2 = r1.with_step 2` I essentially _select every other element_.

This analogy breaks down a bit if I then do `r3 = r2.with_step 1` because after the 'selection' I will have more entries than before 😅 but I think it could work.

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Regression.enso [hunk 7]
```diff
@@ -103,7 +103,8 @@ type Fitted_Model
         Fitted_Model.Logarithmic a b _ -> a.to_text + " * X.ln + " + b.to_text
         Fitted_Model.Power a b _ -> a.to_text + " * X ^ " + b.to_text
 
-    ## Use the model to predict a value.
+    ## GROUP Math
```
Comment (radeusgd) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Data/Regression.enso]: Given it's more complicated, my intuition leans that maybe this should be in `Calculations`?

Comment (jdunkerley) [reply distribution/lib/Standard/Base/0.0.0-dev/src/Data/Regression.enso]:
```suggestion
    ## GROUP Calculations
```

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Range.enso (hunk 6 above)
Comment (jdunkerley) [reply distribution/lib/Standard/Base/0.0.0-dev/src/Data/Range.enso:40]: When we experimented with this it felt right to have the setters in the same category as getter.

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Numbers.enso [hunk 8]
```diff
@@ -1029,7 +1093,8 @@ type Integer
     bit_shift : Integer -> Integer ! Arithmetic_Error
     bit_shift self that = @Builtin_Method "Integer.bit_shift"
 
-    ## Performs a left-wise bit shift on the bits of this.
+    ## GROUP Bitwise
+       Performs a left-wise bit shift on the bits of this.
```
Comment (radeusgd) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Data/Numbers.enso]: Again, not really this PR but the doc changes highlight these.

I was not familiar with the phrase 'left-wise bit shift'. Trying to Google it I find some results but not much. Would a 'bit-wise left shift' be possibly more familiar to users?

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Text/Extensions.enso [hunk 9]
```diff
@@ -1356,7 +1387,8 @@ Text.index_of self term="" start=0 case_sensitivity=Case_Sensitivity.Sensitive =
         span = used.locate term Matching_Mode.First case_sensitivity
         if span.is_nothing then Nothing else span.start+used_start
 
-## Find the last index of the `term` in the input.
+## GROUP Values
+   Find the last index of the `term` in the input.
```
Comment (radeusgd) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Data/Text/Extensions.enso:1391]: loose suggestion - shall we maybe introduce another group like `Search`?

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Text/Regex.enso [hunk 10]
```diff
@@ -149,7 +155,8 @@ type Regex
         Helpers.expect_text input <|
            self.match_all input . map match_to_group_maybe
 
-    ## Splits the `input` text based on the pattern described by `self`.
+    ## GROUP Conversions
+       Splits the `input` text based on the pattern described by `self`.
```
Comment (radeusgd) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Data/Text/Regex.enso:159]: I'm a bit surprised to find this in `Conversions`.

It does make some sense (it is parsing the text into parts, and stuff like parse-to-integer _are_ conversions), but it is not really changing the type (it's still text, just in parts).

Comment (jdunkerley) [reply distribution/lib/Standard/Base/0.0.0-dev/src/Data/Text/Regex.enso:159]: It felt correct to have it with parse (which is a conversion) so split and tokenize ended up here.

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Time/Day_Of_Week.enso [hunk 11]
```diff
@@ -4,22 +4,31 @@ from project.Data.Ordering import all
 
 polyglot java import java.time.DayOfWeek
 
+## Represents a day of the week.
 type Day_Of_Week
+    ## Create a Day_Of_Week indicating Sunday.
```
Comment (radeusgd) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Data/Time/Day_Of_Week.enso]:
```suggestion
    ## Day_Of_Week representing Sunday.
```

I'd rephrase these.

It's not necessarily `Create` because the constructor may also be used in a pattern-match where it is **de**constructing a value, not creating it.

Very subjective: `indicating` sounds to me like about an error (it's the phrasing we use in many error doc comments), `representing` sounds a bit less loaded IMO.

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Time/Day_Of_Week.enso [hunk 12]
```diff
@@ -4,22 +4,31 @@ from project.Data.Ordering import all
...
 
+    ## Create a Day_Of_Week indicating Friday.
     Friday
 
+    ## Create a Day_Of_Week indicating Saturday.
     Saturday
 
-    ## Convert the Day_Of_Week to an Integer
+    ## GROUP Conversions
+       Convert the Day_Of_Week to an Integer
```
Comment (radeusgd) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Data/Time/Day_Of_Week.enso]:
```suggestion
       Convert the Day_Of_Week to an Integer.
```

we could fill-in the missing dot while we are at it :)

File: distribution/lib/Standard/Base/0.0.0-dev/src/Math.enso [hunk 13]
```diff
@@ -1,9 +1,7 @@
 import project.Data.Numbers.Decimal
 import project.Data.Numbers.Number
 
-## Alias Pi (Constant)
-
-   The mathematical constant pi, equal to the ratio of a circle circumference
+## The mathematical constant pi, equal to the ratio of a circle circumference
```
Comment (radeusgd) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Math.enso:4]: Shouldn't these constants get a `GROUP`? I assume `Math` would fit just right.

Comment (jdunkerley) [reply distribution/lib/Standard/Base/0.0.0-dev/src/Math.enso:4]: Deliberately not so they don't end up too high in the static list.

File: distribution/lib/Standard/Base/0.0.0-dev/src/Meta.enso [hunk 14]
```diff
@@ -226,6 +236,7 @@ Any.is_same_object_as : Any -> Boolean
 Any.is_same_object_as self value = is_same_object self value
 
 ## ADVANCED
+   GROUP Logical
```
Comment (radeusgd) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Meta.enso]: nitpick: but wondering if this fits `Logical`, also typechecks are a pretty advanced feature from perspective of GUI usesrs - I'm even wondering if we should display them in the GUI. Maybe they should get a separate group `Types`? If not, maybe `Metadata` would fit here more?

File: distribution/lib/Standard/Base/0.0.0-dev/src/Panic.enso [hunk 15]
```diff
@@ -228,7 +227,8 @@ type Caught_Panic
     convert_to_dataflow_error : Error
     convert_to_dataflow_error self = @Builtin_Method "Caught_Panic.convert_to_dataflow_error"
 
-    ## Returns the stack trace of the caught panic.
+    ## GROUP Errors
```
Comment (radeusgd) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Panic.enso]: Looking at it, I guess this definitely needs an `ADVANCED` tag

```suggestion
    ## ADVANCED
       GROUP Errors
```

File: distribution/lib/Standard/Base/0.0.0-dev/src/System/File.enso [hunk 16]
```diff
@@ -305,7 +308,8 @@ type File
     / : (Text | File) -> File
     / self subpath = self.resolve subpath
 
-    ## Join two or more path segments together, normalizing the `..` and `.` subpaths.
+    ## GROUP Metadata
+       Join two or more path segments together, normalizing the `..` and `.` subpaths.
```
Comment (radeusgd) [inline distribution/lib/Standard/Base/0.0.0-dev/src/System/File.enso]: Why is that `Metadata`?

The definition of the group says: `Metadata:  A property of the object (e.g. length, row_count).`. I see nothing like that here.

It seems more like a transformation, maybe it should be `Calculations` or `Selections`?

File: distribution/lib/Standard/Base/0.0.0-dev/src/System/File_Format.enso [hunk 17]
```diff
@@ -110,7 +112,9 @@ type File_Format
         make_name type_obj = type_obj.to_text.replace "_Format" "" . replace "_" " "
         Single_Choice display=Display.Always values=(all_types.map n->(Option (make_name n) (File_Format.constructor_code n)))
 
+## A file format for plain text files.
 type Plain_Text_Format
+    ## Create a file format for plain text files with the specify encoding.
```
Comment (radeusgd) [inline distribution/lib/Standard/Base/0.0.0-dev/src/System/File_Format.enso]: I'd rephrase this, the current state sounds off grammatically, at least IMO.
```suggestion
    ## Create a file format for plain text files with the specified encoding.
```

File: distribution/lib/Standard/Database/0.0.0-dev/src/Data/Column.enso [hunk 18]
```diff
@@ -766,6 +792,7 @@ type Column
         result.cast Value_Type.Float . rename new_name
 
     ## ALIAS int
+       GROUP Standard.Base.Math
```
Comment (radeusgd) [inline distribution/lib/Standard/Database/0.0.0-dev/src/Data/Column.enso]: The same operation in `Numbers` was in `GROUP Rounding`. I think we need to be consistent.

```suggestion
    ## ALIAS int
       GROUP Standard.Base.Rounding
```

File: distribution/lib/Standard/Database/0.0.0-dev/src/Data/Column.enso [hunk 19]
```diff
@@ -786,7 +813,8 @@ type Column
             new_name = self.naming_helper.function_name "truncate" [self]
             self.make_unary_op "TRUNCATE" new_name
 
-    ## Takes the ceiling of floating-point values, returning integer values.
+    ## GROUP Standard.Base.Math
```
Comment (radeusgd) [inline distribution/lib/Standard/Database/0.0.0-dev/src/Data/Column.enso]: I think rounding related operations on Column need to be moved.
```suggestion
    ## GROUP Standard.Base.Rounding
```

File: distribution/lib/Standard/Database/0.0.0-dev/src/Data/Column.enso [hunk 20]
```diff
@@ -1103,7 +1147,8 @@ type Column
         new_name = self.naming_helper.function_name "ends_with" [self, other]
         make_text_case_op self "ends_with" other case_sensitivity new_name
 
-    ## Checks for each element of the column if it contains `other`.
+    ## GROUP Standard.Base.Logical
+       Checks for each element of the column if it contains `other`.
```
Comment (radeusgd) [inline distribution/lib/Standard/Database/0.0.0-dev/src/Data/Column.enso:1151]: Inconsistency.

I'm pretty sure that `starts_with`, `ends_with` and `contains` should belong to the same group.	But this one is in `Logical` whereas `ends_with` is in `Text`. IMO all of these belong to `GROUP Text`.

File: distribution/lib/Standard/Database/0.0.0-dev/src/Data/Column.enso [hunk 21]
```diff
@@ -1119,7 +1164,8 @@ type Column
         new_name = self.naming_helper.function_name "contains" [self, other]
         make_text_case_op self "contains" other case_sensitivity new_name
 
-    ## Checks for each element of the column if it matches an SQL-like pattern.
+    ## GROUP Standard.Base.Logical
+       Checks for each element of the column if it matches an SQL-like pattern.
```
Comment (radeusgd) [inline distribution/lib/Standard/Database/0.0.0-dev/src/Data/Column.enso:1168]: This also seems to be more fitting for the `Text` group.

File: distribution/lib/Standard/Table/0.0.0-dev/src/Data/Column.enso [hunk 22]
```diff
@@ -141,7 +141,8 @@ type Column
         IO.println (self.display show_rows format_terminal=True)
```
Comment (radeusgd) [inline distribution/lib/Standard/Table/0.0.0-dev/src/Data/Column.enso:141]: I'm getting worried that with so many entries and group assignments, we can easily get the in-memory and Database Column/Table out-of-sync.

Do you think we should include a 'pass' in your doc-comment linter which would verify that the corresponding entries are consistent with each other?

File: distribution/lib/Standard/Table/0.0.0-dev/src/Data/Expression.enso [hunk 23]
```diff
@@ -4,6 +4,7 @@ polyglot java import java.lang.IllegalArgumentException
 polyglot java import java.lang.UnsupportedOperationException
 polyglot java import org.enso.table.expressions.ExpressionVisitorImpl
 
+## Functions for parsing and handling Enso expressions.
```
Comment (radeusgd) [inline distribution/lib/Standard/Table/0.0.0-dev/src/Data/Expression.enso]: The only method of this type is `PRIVATE`, shall we make the type `PRIVATE` as well?

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Text/Regex.enso (hunk 10 above)
Comment (radeusgd) [reply distribution/lib/Standard/Base/0.0.0-dev/src/Data/Text/Regex.enso:159]: I see, ok.

File: distribution/lib/Standard/Base/0.0.0-dev/src/Math.enso (hunk 13 above)
Comment (radeusgd) [reply distribution/lib/Standard/Base/0.0.0-dev/src/Math.enso:4]: Ah. I assume the `Constants` group would have the same problem?

File: distribution/lib/Standard/Table/0.0.0-dev/src/Data/Type/Value_Type.enso [hunk 24]
```diff
@@ -140,13 +141,15 @@ type Value_Type
        In-Memory and SQLite tables support this.
     Mixed
 
-    ## Checks if the `Value_Type` represents a boolean type.
+    ## GROUP Standard.Base.Metadata
+       Checks if the `Value_Type` represents a boolean type.
```
Comment (radeusgd) [inline distribution/lib/Standard/Table/0.0.0-dev/src/Data/Type/Value_Type.enso:145]: I'm wondering if these need a group. They are really mostly for the internal use to check if the type is fitting. I'd consider removing the GROUP and adding `PRIVATE ADVANCED` pair.

Comment (radeusgd) [reply distribution/lib/Standard/Table/0.0.0-dev/src/Data/Type/Value_Type.enso:145]: But also fine with keeping as-is. I imagine these methods _can_ be useful to end-users and they are not magic that needs to be hidden.

File: distribution/lib/Standard/Table/0.0.0-dev/src/Errors.enso [hunk 25]
```diff
@@ -41,7 +41,7 @@ type Too_Many_Column_Names_Provided
     ## PRIVATE
        More names than the column count provided to the function.
        Can occur when using By_Position.
-    Error (column_names : [Text])
+    Error (column_names : Vector Text)
```
Comment (radeusgd) [inline distribution/lib/Standard/Table/0.0.0-dev/src/Errors.enso:44]: Wow that is an ancient artifact you found here!

Comment (jdunkerley) [reply distribution/lib/Standard/Table/0.0.0-dev/src/Errors.enso:44]: Yep - I was thinking about adding a check to linter for this style of type.

File: distribution/lib/Standard/Base/0.0.0-dev/src/Math.enso (hunk 13 above)
Comment (jdunkerley) [reply distribution/lib/Standard/Base/0.0.0-dev/src/Math.enso:4]: Yep Constants ends up at the top.

File: distribution/lib/Standard/Base/0.0.0-dev/src/Meta.enso (hunk 14 above)
Comment (jdunkerley) [reply distribution/lib/Standard/Base/0.0.0-dev/src/Meta.enso]: Removed from GROUP so they end up lower down.

**Review by radeusgd — APPROVED** [review]
Looks great, I added some suggestions above.
